### ISSUE TITLE
0.4.0 release in 3Q 2020

### ISSUE THREAD

**Issue by brodybits** [xmldom/xmldom#99]
__updated:__

I think we should make the new 0.4.0 release soon and adjust the planning of the following milestones:

- https://github.com/xmldom/xmldom/milestone/1 (0.4.0) - push remaining items back to 0.5.0
- https://github.com/xmldom/xmldom/milestone/3 (0.5.0) - consider moving these into a 0.6.0 release

__Added item: update changelog (issue #62)__

~~I would like to get a couple more changes in before making the release, if they don't take too long:~~

- ~~PR #36 (cleaner module exports in index.js)~~
- ~~fix main package entry (issue #53)~~

/cc @karfau @kethinov @timbru31 please speak up asap in case of any concerns

**Comment by karfau** [comment]
DO I understand correctly, that the only changes you want to have as part of 0.4.0 are the ones listed in this description and the other open ticket would be moved to "a new 0.5.0"?

I am working on a much better test coverage on my machine using jest, there shouldn't be much stuff to do to have a PR ready. It's up to you to say whether this should be "part of the release" or it's not related.
I'm also able to invest more time again from this week on (my relocation and vacation is "complete").

**Comment by brodybits** [comment]
That would be really awesome if you have the improved test coverage in progress. I would also not mind starting with an incomplete, draft PR if you like.

I am actually pretty busy with some other project in parallel so there is not a really super rush on my part. I was just triggered by both the number of months and the number of valuable fixes that have been made since the last release.

**Comment by karfau** [comment]
I think I can commit to file at least a draft PR latest at the weekend. I will describe the issues I had with this approach in the PR then.

**Comment by brodybits** [comment]
FYI some recent contributions do not seem to be very well tested in the mutation report:

- getElementsByClassName - updated by PR #91
- code affected by PR #100

**Comment by brodybits** [comment]
I think we should try to make this release soon. It would be ideal if we could update the changelog as discussed in issue #62.

**Comment by brodybits** [comment]
@karfau I would like to make the release this week if there is nothing else missing.

**Comment by karfau** [comment]
When you say "this week", do you mean until today or until 4th of October?
In the latter case  I'll try to finish the PR related to the Changelog asap.
I'm not able to commit to any thing relevant to the actual behavior of xmldom within this week.

**Comment by brodybits** [comment]
By "this week" I had meant before 3rd of October, if possible. (I had actually been hoping to make the release before 1 October.) But let's make it 5th October at the latest. I will look forward to the updated Changelog.

**Comment by brodybits** [comment]
@karfau I would like to make this release Monday evening/night. I just put this on my own calendar to avoid slipping again. I will probably just copy the CHANGELOG.md updates from PR #138 when I make this release. Any final objections?

**Comment by karfau** [comment]
@brodybits Not a single one.

PS: I think we should try to automate more parts of the release process, so it's not such a burden on you. But this is of course not a simple task.

**Comment by brodybits** [comment]
I have made the release with CHANGELOG.md updated based on PR #138, published on npm, put the CHANGELOG info into the release on GitHub, and bumped the version to `0.4.1-dev`. I have moved all remaining 0.4.0 milestone items to the 0.5.0 milestone. I think this 0.4.0 release is now done.

I did also discover today https://github.com/ai/dual-publish via Twitter, and it seems to use release-it. I wonder if we should investigate and consider using these 2 tools.
